feat: add password reset process pages

Maaayhan opened the pull request:
Resolves CP-121

[CP-121](https://beequant.atlassian.net/browse/CP-121)

Have to merge **CP-28 branch** of platform_api firstly before merging this branch

---

## What happens here?

- Implemented "Forgot Password" and "Reset Password" forms
- Created a Success Reset page to confirm successful password reset
- Added appropriate user guidance messages throughout the reset process
- Updated AuthFormContainer for reuse
- Added multilingual support (modified the Link references in the Route file to avoid navigation bugs)
- Added passwordSchema in 'utils' for reuse

Resolves CP-121

### UI comparison
**Legacy:**
<img width="439" alt="image" src="https://github.com/user-attachments/assets/94f8b7da-6fd9-457e-bf56-2a9ffb8c7731">

**Turbo:**
<img width="540" alt="image" src="https://github.com/user-attachments/assets/2bfe7b99-04dc-4b06-97f2-a28b34c6b42c">

**Legacy:**
<img width="540" alt="image" src="https://github.com/user-attachments/assets/bcd5c0a6-1b3f-4be1-8395-123abaa19fe7">

**Turbo:**
<img width="600" alt="image" src="https://github.com/user-attachments/assets/c2405e9f-1be4-4b80-a511-707cd67e9553">

**Legacy:**
<img width="539" alt="image" src="https://github.com/user-attachments/assets/4f5e55e5-aa1d-43ff-83b1-6cbf44e1a841">

**Turbo:**
<img width="546" alt="image" src="https://github.com/user-attachments/assets/e5fbe4ac-d4dd-469b-a3e0-f6dcbd84cc79">


## How do I know this is working?
- For testing purposes, please switch the backend to the branch ‘**feat/CP28-reset-user-password**’ in platform_api repo.
- When the user clicks the "Forgot password" text on the login page, they are directed to the Reset Password Initiation Page, which will prompt them to enter their registered email address.
![image](https://github.com/user-attachments/assets/88222198-5e9a-4172-8a81-2c9fa16660e4)
- After submission, the system will display the message "If the email is associated with an account, a reset email will be sent shortly. Please check your mailbox," guiding the user to check their email for the reset link. If an error occurs during the request, the message "An error occurred while processing your request. Please try again later." will be shown.
![image](https://github.com/user-attachments/assets/38a0eb43-461c-4ec5-9681-2ee13e016157)

- Clicking the link in the email will navigate the user to the Reset Password Page, where they can enter a new password.
For testing purposes, please use the GraphQL playground at http://localhost:3000/graphql with the following query to get 'resetPasswordToken'：
```bash
  getUsers {
    id
    email
    resetPasswordToken
  }
}
```
And then manually visit [http://localhost:5173/reset-password?token={resetPasswordToken}](http://localhost:5173/reset-password?token=%7BresetPasswordToken%7D).
![image](https://github.com/user-attachments/assets/7d863c79-db62-43ca-a9e6-0ca7fd6151fb)

- If the reset is successful, the user will be navigated to the reset success page. On this page, the user can directly click the "Back to Login" button to log in on the Login Page.
![image](https://github.com/user-attachments/assets/1cd114e7-35a9-45db-9d6b-b9eb5508ca34)

- During the process, users can always click the "Login" text at the bottom to return to the login page.
<img width="529" alt="image" src="https://github.com/user-attachments/assets/d640bba7-a63c-4cbd-8d4b-f382d4366e9b">

## Any notes?

<!--
  New PR Checklist

- Have you tag the right reviewer?
- Have you dm the reviewer?
- Maybe @ the reviewer in channel just to be safe

- is CI passing in your pr?
- Have you update the ticket with the right status?
-->

Commit:
refactor login form

Commit:
style login form as per style guide

Commit:
flip user loading state in platform app

Commit:
more styling changes to login form

Commit:
better lookking theme toggle

Commit:
style register form

Commit:
fix input focused style

s3713434 committed:
feat: Optimize checkbox and add background colors to Tailwind config

Fixed the bug where the label and checkbox were not changing color simultaneously, added predefined background colors to the Tailwind config, and set the font size to 13px to match the previous frontend framework.

Resolve CP-97

Commit:
add more types to login form

s3713434 committed:
feat: Adding third-party section on the Login Page

Added a third-party section(buttons)
- implemented third-party section

Resolve CP-100

Commit:
merge in latest platform_app

Commit:
fix login form field not linked correctly

Commit:
add new icons for dashboard menu items

Commit:
add dashboard layout

Commit:
break out sidebar data n stuffs

Commit:
organize sidebar menu item styles

Commit:
add mobile compatibility to sidebar

Commit:
fix sidebar not overflowing when in md viewport

Commit:
add logout action in auth module

Commit:
connect dashboard link in sidebar

Commit:
fix dashboard layout content styling

Commit:
add profile menu

Commit:
more fix to dashboard layout styling

s3713434 committed:
Fixed the text-size problem

Commit:
add pull request template

Commit:
discard old apps from repo

Commit:
fix ci errors

Commit:
update ci triggers

Commit:
add turbo cache to ci

Commit:
add pr checklist

Commit:
regen yarn.lock

s3713434 committed:
Change a bit style

s3713434 committed:
Change the button color

s3713434 committed:
Update: Create Social media button

s3713434 committed:
Added primary and accent color palettes, and subsequently updated the styles for the login form, register form, buttons, checkboxes, and layout.

Commit:
Merge pull request #3 from BeeQuantAI/feat/dashboard-layout

feat: migrate dashboard layout

s3713434 committed:
Merge branch 'feat/CP-100-adding-third-party-section' into CP-102-optimize-login-and-register-form

s3713434 committed:
change the color(?

s3713434 committed:
Merge branch 'dev' into CP-102-optimize-login-and-register-form

s3713434 committed:
Change the dashboard color(?

s3713434 committed:
Update: Solved most of problems, will working on the color now

s3713434 committed:
adding back the text color

Commit:
prettify entire repo

Commit:
handle unrecorded auth error

Commit:
move dashboard layout stuffs around

Commit:
turn off next image warning

Commit:
add profile card

s3713434 committed:
Changed the color scale

s3713434 committed:
Merge branch 'CP-102-optimize-login-and-register-form' of https://github.com/BeeQuantAI/beequant-turbo into CP-102-optimize-login-and-register-form, and modfied the color scale

Commit:
refactor dashboard layout with new colors

Commit:
remove pgadmin from backend docker compose

Commit:
fix sidebar button hover effect

Commit:
break out tailwind colors

Commit:
Merge pull request #5 from BeeQuantAI/feat/migrate-profile-page

feat/migrate profile page

Commit:
update README.md with new instruction to start up client

Commit:
refactor social button a bit to restrict it's useage

Commit:
Merge remote-tracking branch 'origin/dev' into CP-102-optimize-login-and-register-form

Commit:
fix import error in login form

Commit:
fix props type in social button

Commit:
fix checkbox having fixed width

Commit:
more color fixes

Commit:
Merge pull request #4 from BeeQuantAI/CP-102-optimize-login-and-register-form

Cp 102 optimize login and register form

Commit:
Merge pull request #1 from BeeQuantAI/feat/CP-97-optimize-checkbox

feat: Optimize checkbox and add background colors to Tailwind config

Commit:
overhaul theming in login form

s3713434 committed:
create the 200 page

s3713434 committed:
Merge branch 'dev' of https://github.com/BeeQuantAI/beequant-turbo into feat/cp-107-create-registration-success-page

s3713434 committed:
Merge branch 'main' into feat/cp-107-create-registration-success-page

Commit:
oops

99032000 committed:
clean dependency

Commit:
able to start both client and server in prod

Commit:
downgrade next to 14.1.4

Commit:
refresh yarn.lock

Commit:
downgrade yarn to 1.22.22

Commit:
Merge remote-tracking branch 'origin/dev' into feat/theming-reimagined

Commit:
refresh yarn.lock

Commit:
add missing import in tailwind config

Commit:
refactor auth forms

Commit:
fix button having fixed width

Commit:
slow down global animation transition

Commit:
refactor controlled text input with error message

Commit:
tweak input error message color

Commit:
restyle form error message

Commit:
fix password input not hiding password

Commit:
overhaul dashboard layout styling

Commit:
restyle profile card

Commit:
Merge pull request #7 from BeeQuantAI/feat/theming-reimagined

feat/theming reimagined

Commit:
Merge remote-tracking branch 'origin/dev' into feat/cp-107-create-registration-success-page

Commit:
rerun graphql codegen in client

Commit:
add simple message about empty root page

Commit:
move theme switcher from root layout to auth layout

Commit:
move theme switcher from root layout to auth layout

s3713434 committed:
change back the register action

s3713434 committed:
Merge branch 'feat/cp-107-create-registration-success-page' of https://github.com/BeeQuantAI/beequant-turbo into feat/cp-107-create-registration-success-page

s3713434 committed:
Delete the await at auth-service.ts no need for it in redirect.

s3713434 committed:
Merge pull request #6 from BeeQuantAI/feat/cp-107-create-registration-success-page

Feat/cp 107 create registration success page

Commit:
stop tracking yarn file

Commit:
migrate identical styling from old platform app

Commit:
move pr template to a level up

Commit:
add dark mode to account-menu

Commit:
supress inconsistent attribute warning on html tag

Commit:
remove conflicting favicon file

Commit:
update accent color palette

s3713434 committed:
feat: add custom font size configuration in Tailwind CSS theme

s3713434 committed:
chore: update existing font sizes to match custom configuration in Tailwind CSS

SeijaHuang committed:
Merge branch 'dev' of https://github.com/BeeQuantAI/beequant-turbo into dev

Commit:
Feat: Optimized the README file

Optimized the README file:
- Optimized the `## How do I run this`

Resolve: Under CP-106

Chris-Wang committed:
Merge pull request #12 from BeeQuantAI/feat/Optimize-README-File

Feat: Optimized the README file

Commit:
fix sidebar menu accordion hover styles

Commit:
tweak accent palette color

Commit:
Merge pull request #11 from BeeQuantAI/cp-104-organize-and-sort-fonts-from-legacy-repo-by-size

Cp 104 organize and sort fonts from legacy repo by size

Commit:
Merge remote-tracking branch 'origin/dev' into feat/cp-111-dashboard-style-tweak

SeijaHuang committed:
Merge branch 'dev' of https://github.com/BeeQuantAI/beequant-turbo into dev

SeijaHuang committed:
feat: add setting page
Added account setting page:
- implemented account setting form
- implemented account setting page

Added variant for shared input component:
- added display variant to enable the label and input displaying either horizontally or vertically

Added user help messages and validation messages:
- added user help messages to automatically pop up when clicking responding input field
- added validation messages which are mainly for form validation error messages

Resolve CP-108

Commit:
more dashboard tweaks

Chris-Wang committed:
Merge pull request #9 from BeeQuantAI/feat/CP-108-add-setting-page

Feat/cp 108 add setting page

Commit:
hardcode line-height in sidebar menu

Commit:
regress one stop dev solution to make it compatible with windows env

Commit:
update readme with new instruction to start local dev env

Commit:
clean up readme

s3713434 committed:
refactor: match icon colors with old repo and clean up input control font sizes

Commit:
Merge branch 'dev' into feat/cp-111-dashboard-style-tweak

Commit:
Merge pull request #10 from BeeQuantAI/feat/cp-111-dashboard-style-tweak

feat/cp 111 dashboard style tweak

s3713434 committed:
refactor: convert existing register-succeed page into a reusable component

99032000 committed:
clean dependency

s3713434 committed:
feat: add responsive design to registration and login pages

floraSimpleDev committed:
fix: Replace Icons to Match Legacy Repo Styles

Fixed the Show Password Icon Animation

Resolve CP-116

fix/CP-116-Replace-Icons-to-Match-Legacy-Repo-Styles

s3713434 committed:
feat: implement ultrawide screen responsiveness for login, registration, and 200 pages

s3713434 committed:
Yeah change back the font-size

s3713434 committed:
change the min-w to max-w :3 it should be good now

SeijaHuang committed:
Feat: add internationalization
Add app setting page:
- Implemented language change
- Implemented theme change
Add internationalization
- Add next-intl library to achieve internationalization routing
- Modify middleware with next-intl middleware
- Modify next navigations with next-intl shared routing, including redirect, useRouter etc.
Resolve CP-113

NorrisWu0 commented on apps/client/src/module/auth/forgot-password-form.tsx:
```diff
@@ -0,0 +1,81 @@
+    .string()
+    .min(1, { message: EmailErrorMsgs.Required })
+    .email({ message: EmailErrorMsgs.Invalid })
+    .regex(emailPattern, { message: EmailErrorMsgs.Invalid }),
```
Same comment as in the other pr.

NorrisWu0 commented on apps/client/src/module/auth/reset-password-form.tsx:
```diff
@@ -0,0 +1,110 @@
+type FormSchema = z.infer<typeof formSchema>;
+const formSchema = z
+    .object({
+        password: z.string()
```
Maybe this can be broken out into a password schema object in a util somewhere?

NorrisWu0 commented on apps/client/src/module/auth/reset-password-form.tsx:
```diff
@@ -0,0 +1,110 @@
+    const [token, setToken] = useState('');
+
+    useEffect(() => {
+        const urlParams = new URLSearchParams(window.location.search);
```
there's a different way to get url Params. You probably want to check the docs on NextJS.

NorrisWu0 replied:
https://nextjs.org/docs/app/api-reference/file-conventions/page#searchparams-optional

NorrisWu0 commented on apps/client/src/module/auth/reset-password-form.tsx:
```diff
@@ -0,0 +1,110 @@
+            setResponseMessage(res?.error || UserHelpMsgs.ForgotPasswordError);
+        }
+        else {
+            setResponseMessage("Processing...");
```
Oh...? How does this work? Do you need to invoke the action again to set the proper response?

NorrisWu0 commented on apps/client/src/module/auth/reset-password-form.tsx:
```diff
@@ -0,0 +1,110 @@
+
+    return (
+        <AuthFormContainer onSubmit={onSubmit} error={errors.root?.message}  title="Reset Password">
+            {responseMessage && (
```
We can probably bake this response into the `<AuthFormContainer/>` too if this gets used much.

NorrisWu0 commented on apps/client/src/utils/user-help-messgage.ts:
```diff
@@ -1,5 +1,8 @@
+import { ForgotPasswordDocument } from './../module/graphql/codegen/graphql';
```
What's the import for?

s3713434 committed:
problem solved, just misunderstanding. xp

s3713434 committed:
Merge branch 'dev' into cp-114-add-responsive-design-to-registeration-and-login-page

s3713434 committed:
Merge pull request #17 from BeeQuantAI/cp-114-add-responsive-design-to-registeration-and-login-page

feat: add responsive design to registration and login pages

SeijaHuang committed:
Feat: add account management and change password page

Add account management page：
- Implemented account management page
- Added account management menu on sidebar and account drop down
Add change password page:
- Implemented change password page with responding mutation
- Added Modal component
Relocate account setting page
- Relocate account setting page with newly added two pages in account folder which is same level with dashboard folder
Resolve CP-125

Commit:
Feat: Migrate landing page  into turbo repo

Completed the following changes:
Client:
- Add the landing page to /app named landing
- Split the landing page to eight operational components to module folder named landing
- Added the images the landing page required into public folder
- Added css files into /app/assets/css
- Imported css files into globals.css
- Changed the redirection logic in middleware.ts, if the token is not exist in the cookies, it will redirect to landing page
- Set the internationalization for landing page

Resolve CP-106

Commit:
use turbopack in client app

Maaayhan committed:
feat: add password reset process pages

- Implemented "Forgot Password" and "Reset Password" forms
- Created a Success Reset page to confirm successful password reset
- Added appropriate user guidance messages throughout the reset process
- Updated AuthFormContainer for reuse
- Added multilingual support (modified the Link references in the Route file to avoid navigation bugs)
- Added passwordSchema in 'utils' for reuse

Resolves CP-121

NorrisWu0 reviewed (dismissed):
don't see any apparent issue, lgtm! feel free to merge
